iris.cabal

```diff
@@ -25,7 +25,7 @@ source-repository head
   location:            https://github.com/chshersh/iris.git
 
 common common-options
-  build-depends:       base >= 4.14 && < 4.17
+  build-depends:       base >= 4.14 && < 4.18
 
   ghc-options:         -Wall
```

Comment:
Let's also add the `-Wredundant-strictness-flags` flag as shown on the screenshot below for GHC 9.4. This is a new GHC sanity-check not enabled by default or by `-Wall` which I still find useful 🙂 

![image](https://user-images.githubusercontent.com/4276606/194521654-476c7c72-3a60-4f20-b0db-d492d87711c6.png)